src/Spryker/Client/RabbitMq/Model/Publisher/Publisher.php

```diff
@@ -240,4 +240,18 @@ protected function getMessageConfig()
     {
         return [];
     }
+
+    /**
+     * @param \Generated\Shared\Transfer\QueueSendMessageTransfer $queueSendMessageTransfer
+     *
+     * @return string|null
+     */
+    protected function getLocale(QueueSendMessageTransfer $queueSendMessageTransfer): ?string
+    {
+        if ($this->config->isDynamicStoreEnabled() === true) {
+            return $queueSendMessageTransfer->getLocale() ?? $this->config->getDefaultLocaleCode();
+        }
+
+        return $queueSendMessageTransfer->getLocale();
+    }
```

Comment:
Why it is done on RabbitMQ level and not in the specific places where we create `QueueSendMessageTransfer`?